src/config-file.c

```diff
@@ -272,11 +274,14 @@ Config* load_config_file(const gchar *config_file, GError **error)
         key_gateway_token_exists = get_key_string(ini_file, "client", "gateway_token",
                                                   &config->gateway_token, NULL, NULL);
         if (!key_auth_token_exists && !key_gateway_token_exists) {
-                g_set_error(error, G_KEY_FILE_ERROR, G_KEY_FILE_ERROR_INVALID_VALUE,
-                            "Neither auth_token nor gateway_token is set in the config.");
-                return NULL;
-        }
-        if (key_auth_token_exists && key_gateway_token_exists) {
+                g_info("Neither auth_token nor gateway_token is set, using client certificates");
+                key_client_cert_exists = get_key_string(ini_file, "client", "client_cert", &config->client_cert, NULL, NULL);
+                key_client_key_exists = get_key_string(ini_file, "client", "client_key", &config->client_key, NULL, NULL);
+                if (!key_client_cert_exists || !key_client_key_exists) {
+                        g_set_error(error, 1, 4, "Neither a token nor client certificate are set!");
+                        return NULL;
+                }
+        } else if (key_auth_token_exists && key_gateway_token_exists) {
```

Comment:
Could you please read these config options outside of these `if` and then add checks for all combinations that are not allowed? This should make the logic simpler here.